tests/glsl-intrinsic/intrinsic-basic.slang

```diff
@@ -0,0 +1,474 @@
+//TEST:SIMPLE(filecheck=CHECK_GLSL): -allow-glsl -stage fragment -entry computeMain -target glsl
```

Comment:
Any reason why we can't use executable test?

Reply:
I couldn't figure how to read back the resulting value and check it for all targets.
At the time of writing this code, if I use RWStructuredBuffer, it was giving me error when the target is glsl or SPIRV.
Please let me know if there is any ways to check if the resulting value is "true".

Reply:
`RWStructuredBuffer` should work, and if it is crashing we should debug it together.

If you merge with top of tree, it should also work if you write SSBO syntax like:

```
//TEST_INPUT:ubuffer(data=[0 0 0 0], stride=4):out,name=outputBuffer
buffer MyBlockName2
{
    float4 result;
} outputBuffer;
```

Reply:
Then you can define the test line as:
```
//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl
```
With that, add these comments to the test file to check the buffer content:
```
// BUF: 1.0
// BUF: 0.0
// BUF: 0.0
// BUF: 0.0
```
This will ensure that the buffer content is `[1.0, 0.0, 0.0, 0.0]`.